backend/ownerSizeTotalWorker.js

```diff
@@ -0,0 +1,277 @@
+  setTimeout(sbp, TASK_TIME_INTERVAL, 'backend/server/computeSizeTaskDeltas')
+})
+
+sbp('sbp/selectors/register', {
+  /**
+   * Selector: 'worker/updateSizeSideEffects'
+   * Handles incoming size update events for a specific resource.
+   * It adds the CID to the temporary persistent index (if not already processed
+   * or pending full recalc) and updates the in-memory delta map (`updatedSizeMap`).
+   *
+   * IMPORTANT: This should only be called for keys where this is relevant,
+   * such as `_private_size_` keys.
+   */
+  'worker/updateSizeSideEffects': async ({ resourceID, size }: { resourceID: string, size: number }) => {
+    // Ignore if the resource is already flagged for a full recalculation.
+    // Its size will be fully computed in `computeSizeTask`.
+    if (updatedSizeList.has(resourceID)) return
+
+    const current = updatedSizeMap.get(resourceID)
+    if (current === undefined) {
+      // First delta update for this resourceID since the last delta task ran.
+      // Add it to the persistent temporary index to ensure it's processed
+      // even if the worker restarts.
+      try {
+        await addToTempIndex(resourceID)
+        // Store the initial delta size.
+        updatedSizeMap.set(resourceID, size)
+      } catch (e) {
+        console.error(`Error adding ${resourceID} to temp index:`, e)
+      }
+    } else {
+      // Accumulate subsequent delta updates in memory.
+      updatedSizeMap.set(resourceID, current + size)
+    }
+  },
+  /**
+   * Selector: 'backend/server/computeSizeTaskDeltas'
+   * Periodically executed task (via setTimeout) to process accumulated
+   * size _deltas_.
+   * Calculates the change in total size for ultimate owners based on the deltas
+   * stored in `updatedSizeMap` and updates the database.
+   */
+  'backend/server/computeSizeTaskDeltas': async function () {
+    // Capture the current set of deltas and clear the map for the next interval
+    const deltaEntries = Array.from(updatedSizeMap)
+    // Clear immediately to start accumulating new deltas.
+    updatedSizeMap.clear()
+
+    // Map to store: ultimateOwnerID -> [totalDelta, Set<originalResourceIDs>]
+    const ultimateOwners = new Map()
+    // Phase 1: Find the ultimate owner for each resource and aggregate deltas.
+    await Promise.all(deltaEntries.map(async ([contractID, delta]) => {
+      let ownerID = contractID
+      // --- Potentially Slow Owner Lookup Loop ---
+      for (let depth = 128; depth >= 0; depth--) {
+        // Fetch the immediate owner
+        const newOwnerID = await sbp('chelonia.db/get', `_private_owner_${ownerID}`, { bypassCache: true })
+        // Found the ultimate owner
+        if (!newOwnerID) break
+        if (!depth) {
+          throw new Error('Exceeded max depth looking up owner for ' + contractID)
+        }
+        ownerID = newOwnerID
+      }
+      // Aggregate delta for the ultimate owner.
+      const [val, set] = ultimateOwners.get(ownerID) || [0, new Set()]
```

Comment:
Could you give this `set` a more descriptive name to describe what it represents?